### PR TITLE
Adopt the binary-compatibility-validator

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: zipline/api/jvm/zipline.api [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+
+public abstract interface class app/cash/zipline/ZiplineService {
+	public abstract fun close ()V
+}
+
+public final class app/cash/zipline/ZiplineService$DefaultImpls {
+	public static fun close (Lapp/cash/zipline/ZiplineService;)V
+}
+
+public abstract interface class app/cash/zipline/internal/bridge/CallChannel {
```
Comment (swankjesse) [inline zipline/api/jvm/zipline.api:344]: Lots of `@PublishedApi` stuff in internal/bridge

File: zipline/api/jvm/zipline.api [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+	public fun toString ()Ljava/lang/String;
+	public abstract fun ziplineFunctions (Lkotlinx/serialization/modules/SerializersModule;)Ljava/util/List;
+}
+
+public final class app/cash/zipline/internal/bridge/ZiplineServiceAdapterKt {
+	public static final fun ziplineServiceAdapter ()Lapp/cash/zipline/internal/bridge/ZiplineServiceAdapter;
+	public static final fun ziplineServiceAdapter (Lapp/cash/zipline/internal/bridge/ZiplineServiceAdapter;)Lapp/cash/zipline/internal/bridge/ZiplineServiceAdapter;
+}
+
+public final class app/cash/zipline/testing/EndpointsKt {
```
Comment (swankjesse) [inline zipline/api/jvm/zipline.api:555]: TODO: move this to testing/

File: zipline/testing/api/testing.api
Comment (swankjesse) [inline zipline/testing/api/testing.api:1]: I don’t know why the plugin generated this for `testing.api` as the plugin isn’t installed for it. Should fix by adopting test facets.

File: zipline-bytecode/src/test/kotlin/app/cash/zipline/bytecode/ApplySourceMapToBytecodeTest.kt [hunk 3]
```diff
@@ -213,11 +212,16 @@ class ApplySourceMapToBytecodeTest {
...
-""".trimMargin()
+      """.trimMargin()
 
     val bytecode = quickJs.compile(js, "demo.js")
     val bytecodeWithSourceMap = applySourceMapToBytecode(bytecode, emptySourceMap)
     quickJs.execute(bytecodeWithSourceMap)
     assertThat(quickJs.evaluate("doubleToDisplayString('1000.0')")).isEqualTo("1000.0")
   }
+
+  @Suppress("INVISIBLE_MEMBER") // Access :zipline internals.
```
Comment (swankjesse) [inline zipline-bytecode/src/test/kotlin/app/cash/zipline/bytecode/ApplySourceMapToBytecodeTest.kt:223]: Hiding `internal` things might be more trouble than it’s worth

Comment (swankjesse) [reply zipline-bytecode/src/test/kotlin/app/cash/zipline/bytecode/ApplySourceMapToBytecodeTest.kt:223]: (Though perhaps adding a testing facet gets me where I want to be.)

File: build.gradle.kts [hunk 4]
```diff
@@ -226,3 +228,14 @@ allprojects {
     environment("ZIPLINE_ROOT", rootDir.toString())
   }
 }
+
+subprojects {
+  plugins.withId("binary-compatibility-validator") {
+    configure<ApiValidationExtension> {
+      // Making this properly internal requires some SQLDelight work.
```
Comment (veyndan) [inline build.gradle.kts:235]: Is there an existing SQLDelight issue for this that we can link?

File: zipline-bytecode/src/test/kotlin/app/cash/zipline/bytecode/ApplySourceMapToBytecodeTest.kt (hunk 3 above)
Comment (veyndan) [reply zipline-bytecode/src/test/kotlin/app/cash/zipline/bytecode/ApplySourceMapToBytecodeTest.kt:223]: Agreed

**Review by veyndan — APPROVED** [review]

File: build.gradle.kts (hunk 4 above)
Comment (swankjesse) [reply build.gradle.kts:235]: It’s work on our end, not theirs. SQLDelight visibility is always public, so we need to encapsulate through some other means.

**Comment by JakeWharton** [comment]
Keep in mind that this is a binary API dumper and NOT actually a binary compatibility validator. Reviewers need to still be vigilant to validate backwards compatibility as part of the review process.

File: zipline/api/jvm/zipline.api (hunk 2 above)
Comment (swankjesse) [reply zipline/api/jvm/zipline.api:555]: https://github.com/cashapp/zipline/pull/1041

**Pull request merged**